pkg/engine/plan.go

```diff
@@ -200,11 +201,11 @@ func (res *planResult) Chdir() (func(), error) {
 // Walk enumerates all steps in the plan, calling out to the provided action at each step.  It returns four things: the
 // resulting Snapshot, no matter whether an error occurs or not; an error, if something went wrong; the step that
 // failed, if the error is non-nil; and finally the state of the resource modified in the failing step.
-func (res *planResult) Walk(cancelCtx *Context, events deploy.Events, preview bool) error {
+func (res *planResult) Walk(cancelCtx *Context, events deploy.Events, preview bool) *result.Result {
 	ctx, cancelFunc := context.WithCancel(context.Background())
 
 	done := make(chan bool)
-	var err error
+	var walkResult *result.Result
```

Comment:
can't call this 'res' because that's the name we're using for this 'this' instance.